src/app/components/courses-page/courses-page.component.ts

```diff
@@ -1,10 +1,31 @@
-import { Component } from '@angular/core';
+import { Component, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+
+import { courses } from 'src/app/shared/mock/courses';
+import { ICourse } from 'src/app/shared/interfaces/course';
 
 @Component({
   selector: 'app-courses-page',
   templateUrl: './courses-page.component.html',
-  styleUrls: ['./courses-page.component.scss']
+  styleUrls: ['./courses-page.component.scss'],
 })
-export class CoursesPageComponent {
+export class CoursesPageComponent implements OnInit, OnChanges {
+  coursesArray: ICourse[] = [];
```

Comment:
While technically correct, it's generally not considered best practice to include the data type or structure
You may just use courses: ICourse[] = [];

Reply:
Ok, got it!